feat: generate action and robot test files that run assert_can_deploy

doudou opened the pull request:
On top of #360

jhonasiv commented on lib/syskit/cli/gen/syskit_app/config/robots/robot.rb.erb:
```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
```
No Robot.config?

doudou replied:
Robot.config is replaced by Robot.setup. `config` was a confusing name from Syskit's perspective. Check out the comment.

If you are confused by this, maybe we should find a way to make it more obvious. Do you have a suggestion ?

jhonasiv replied:
I dont have a problem with the name. It's more that it wasnt clear what were all the possible steps, and since some of the code I've seen used some of the old names (i.e. Robot.config) I wasnt sure what was the exact order of execution of the stages. Maybe add a deprecated warning on the replaced names?

doudou committed:
fix: add test for BulkAssertAtomicActions using example as default arguments

doudou committed:
chore: small simplification

doudou committed:
fix: recursively expand coordination models in profile assertions

doudou committed:
fix: do not terminate early dataflow computations when a port is not updated

The logic was broken in add_port_info, which would reset @changed
to false if the particular port was not updated, leading to the
propagation algorithm to think it found a fixed point.
